go-apps/meep-wais/sbi/wais-sbi.go

```diff
@@ -0,0 +1,256 @@
+		log.Debug("RX MSG: ", mq.PrintMsg(msg))
+		processGisEngineUpdate(msg.Payload)
+	default:
+		log.Trace("Ignoring unsupported message: ", mq.PrintMsg(msg))
+	}
+}
+
+func processActiveScenarioTerminate() {
+	log.Debug("processActiveScenarioTerminate")
+
+	// Sync with active scenario store
+	sbi.activeModel.UpdateScenario()
+
+	sbi.cleanUpCB()
+}
+
+func processActiveScenarioUpdate() {
+
+	log.Debug("processActiveScenarioUpdate")
+
+	formerUeNameList := sbi.activeModel.GetNodeNames("UE")
+
+	sbi.activeModel.UpdateScenario()
+
+	scenarioName := sbi.activeModel.GetScenarioName()
+	sbi.updateScenarioNameCB(scenarioName)
+
+	// Get all UE & POA positions
+	//ueMap, _ := sbi.pc.GetAllUe()
+	poaMap, _ := sbi.pc.GetAllPoa()
+
+	// Update UE info
+	ueNameList := sbi.activeModel.GetNodeNames("UE")
+	for _, name := range ueNameList {
+		ueParent := sbi.activeModel.GetNodeParent(name)
+		if poa, ok := ueParent.(*dataModel.NetworkLocation); ok {
+			apMacId := ""
+			switch poa.Type_ {
+			case mod.NodeTypePoaWifi:
+				apMacId = poa.PoaWifiConfig.MacId
+			}
+			ue := (sbi.activeModel.GetNode(name)).(*dataModel.PhysicalLocation)
+			sbi.updateUeDataCB(name, ue.MacId, apMacId)
+		}
+	}
+
+	//only find UEs that were removed, check that former UEs are in new UE list
+	for _, oldUe := range formerUeNameList {
+		found := false
+		for _, newUe := range ueNameList {
+			if newUe == oldUe {
+				found = true
+				break
+			}
+		}
+		if !found {
+			sbi.updateUeDataCB(oldUe, oldUe, "")
+			log.Info("Ue removed : ", oldUe)
+		}
+	}
+
+	// Update POA Wifi info
+	poaNameList := sbi.activeModel.GetNodeNames(mod.NodeTypePoaWifi)
+	for _, name := range poaNameList {
+		poa := (sbi.activeModel.GetNode(name)).(*dataModel.NetworkLocation)
+		if poa == nil {
+			log.Error("Can't find poa named " + name)
+			continue
+		}
+
+		var longitude *float32
+		var latitude *float32
+		if myPoa, found := poaMap[name]; found {
+			longitude, latitude = parsePosition(myPoa.Position)
+		}
+		//list of Ues MacIds
+		var ueMacIdList []string
+
+		for _, pl := range poa.PhysicalLocations {
+			ueMacIdList = append(ueMacIdList, pl.MacId)
+		}
+		sbi.updateAccessPointInfoCB(name, poa.PoaWifiConfig.MacId, longitude, latitude, ueMacIdList)
+	}
+}
+
+func processGisEngineUpdate(assetMap map[string]string) {
+	for assetName, assetType := range assetMap {
+		// Only process UE updates
+		// NOTE: WAIS might requires distance measurements in the future.
+		//       Not yet implemented, the distance measurements are simply logged here for now.
+		if assetType == postgis.TypeUe {
+			if assetName == postgis.AllAssets {
+				ueMap, err := sbi.pc.GetAllUe()
+				if err == nil {
+					for _, ue := range ueMap {
+						log.Trace("UE[", ue.Name, "] POA [", ue.Poa, "] distance[", ue.PoaDistance, "]")
+					}
+				}
+			} else {
+				ue, err := sbi.pc.GetUe(assetName)
+				if err == nil {
+					log.Trace("UE[", ue.Name, "] POA [", ue.Poa, "] distance[", ue.PoaDistance, "]")
+				}
+			}
+		}
+	}
+}
```

Comment:
distance is not needed. The only thing related to distance is the rssi which is handled in a different way... it can be removed... so I will remove it.